### PR TITLE
Add validations

### PR DESCRIPTION
**Pull request by Ismailco**
- [ ] Add the following validations:
  - [ ] For the User model:
     - [ ] Name must not be blank.
     - [ ] PostsCounter must be an integer greater than or equal to zero.
  - [ ] For the Post model:
     - [ ] Title must not be blank.
     - [ ] Title must not exceed 250 characters.
     - [ ] CommentsCounter must be an integer greater than or equal to zero.
     - [ ] LikesCounter must be an integer greater than or equal to zero.
- [ ] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+RSpec.describe Post, type: :model do
+  user = User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+  subject { Post.new(author: user, title: 'Title', text: 'This is a text') }
+  # it { should belong_to(:author).class_name('User') }
+  # it { should have_many(:comments).foreign_key(:post_id) }
+  # it { should have_many(:likes).foreign_key(:post_id) }
+  # it { should validate_presence_of(:title) }
+  # it { should validate_length_of(:title).is_at_most(250) }
+  # it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+  # it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
```
Comment (medaminedev66) [inline spec/models/post_spec.rb]: - [x] Please, do not commit old pieces of code as inline comments. They will make your project look messy. If you need to review a previous version of your code, you can always use git history.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+      expect(subject.valid?).to be false
+    end
+
+    it 'returns false if the likes conter is not an integer' do
+      subject.title = 'Ismail'
+      subject.likes_counter = 'rr'
+      subject.comments_counter = 1
+      expect(subject.valid?).to be false
+    end
+  end
```
Comment (medaminedev66) [inline spec/models/post_spec.rb:88]: - [x] Any custom methods in the Post model should have a corresponding unit test. Please add a unit test for the Post the update_posts_counter.

File: spec/models/post_spec.rb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+#       expect(subject.valid?).to be false
+#     end
+
+#     it 'returns true if the user name is present' do
+#       subject.name = 'Ismail'
+#       subject.posts_counter = 1
+#       expect(subject.valid?).to be true
+#     end
+#   end
+# end
```
Comment (medaminedev66) [inline spec/models/post_spec.rb]: - [x] Please remove the commented codes to keep your code clean.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      expect(subject.valid?).to be false
+    end
+
+    it 'returns false if the post conter is not an integer' do
+      subject.name = 'Ismail'
+      subject.posts_counter = 'rr'
+      expect(subject.valid?).to be false
+    end
+  end
+end
```
Comment (medaminedev66) [inline spec/models/user_spec.rb:51]: - [x] Following the project requirements, you should also add unit tests for `Comment`, and `Like` models.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @Ismailco  😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- `User` and `Post` models have validations

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — APPROVED** [review]
# Status: Approved :tada: :tada: 

Hi @Ismailco , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**